src/components/course/data/tests/hooks.test.jsx

```diff
@@ -369,87 +299,84 @@ describe('useCourseEnrollmentUrl', () => {
 });
 
 describe('useUserHasSubsidyRequestForCourse', () => {
-  afterEach(() => jest.clearAllMocks());
-
+  const Wrapper = ({ children }) => (
+    <AppContext.Provider value={mockAuthenticatedUser}>
+      {children}
+    </AppContext.Provider>
+  );
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useBrowseAndRequest.mockReturnValue({
+      data: {
+        configuration: undefined,
+        requests: {
+          subscriptionLicenses: [],
+          couponCodes: [],
+        },
+      },
+    });
+  });
   it('returns false when `subsidyRequestConfiguration` are not set', () => {
-    const context = {
-      subsidyRequestConfiguration: null,
-    };
-    const wrapper = ({ children }) => (
-      <SubsidyRequestsContext.Provider value={context}>{children}</SubsidyRequestsContext.Provider>
-    );
-    const { result } = renderHook(() => useUserHasSubsidyRequestForCourse(), { wrapper });
+    const { result } = renderHook(() => useUserHasSubsidyRequestForCourse(), { wrapper: Wrapper });
     expect(result.current).toBe(false);
   });
 
   it('returns false when `subsidyType` is undefined', () => {
-    const context = {
-      subsidyRequestConfiguration: {
-        subsidyRequestsEnabled: true,
-        subsidyType: undefined,
-      },
-    };
-    const wrapper = ({ children }) => (
-      <SubsidyRequestsContext.Provider value={context}>{children}</SubsidyRequestsContext.Provider>
-    );
-    const { result } = renderHook(() => useUserHasSubsidyRequestForCourse(), { wrapper });
-
+    const { result } = renderHook(() => useUserHasSubsidyRequestForCourse(), { wrapper: Wrapper });
     expect(result.current).toBe(false);
   });
 
   it('returns true when `subsidyType` is LICENSE && 1 license request is found', () => {
-    const context = {
-      subsidyRequestConfiguration: {
-        subsidyRequestsEnabled: true,
-        subsidyType: SUBSIDY_TYPE.LICENSE,
-      },
-      requestsBySubsidyType: {
-        [SUBSIDY_TYPE.LICENSE]: [{ state: SUBSIDY_REQUEST_STATE.REQUESTED }],
-        [SUBSIDY_TYPE.COUPON]: [],
+    useBrowseAndRequest.mockReturnValue({
+      data: {
+        configuration: {
+          subsidyRequestsEnabled: true,
+          subsidyType: SUBSIDY_TYPE.LICENSE,
+        },
+        requests: {
+          subscriptionLicenses: ['test-license'],
```

Comment:
nit: while it probably doesn't matter for the test, would recommend having this be an array of objects to reflect the actual data representation. E.g., `subscriptionLicenses: [{ uuid: 'test-license-uuid' }],`